### PR TITLE
Add bioconda recipe for bam2fasta package

### PR DESCRIPTION
**Pull request by pranathivemuri**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/bam2fasta/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set version = "1.0.1" %}
+
+package:
+  name: bam2fasta
+  version: {{ version }}
+
+source:
+  url: https://files.pythonhosted.org/packages/source/b/bam2fasta/bam2fasta-1.0.1.tar.gz
```
Comment (luizirber) [inline recipes/bam2fasta/meta.yaml]: If you set the URL as
```suggestion
  url: https://pypi.io/packages/source/b/bam2fasta/bam2fasta-{{ version }}.tar.gz
```
I think the `autobump` bot will be able to create an update PR when a new version is released to PyPI.

File: recipes/bam2fasta/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - setuptools_scm
+    - setuptools_scm_git_archive
+
+  run:
+    - python
+    - pathos==0.2.5
+    - pysam==0.15.3
+    - screed==1.0
+    - tqdm==4.36.1
+
```
Comment (luizirber) [inline recipes/bam2fasta/meta.yaml]: numpy is missing as a dependency here

**Comment by pranathivemuri** [comment]
@bioconda/core please review!

File: recipes/bam2fasta/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{% set version = "1.0.1" %}
+
+package:
+  name: bam2fasta
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/b/bam2fasta/bam2fasta-1.0.1.tar.gz
```
Comment (bgruening) [inline recipes/bam2fasta/meta.yaml]: you can use the version variable here as well

File: recipes/bam2fasta/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  host:
+    - python
+    - pip
+    - setuptools >=38.6.0
+    - setuptools_scm
+    - setuptools_scm_git_archive
+
+  run:
+    - python
+    - pathos==0.2.5
```
Comment (bgruening) [inline recipes/bam2fasta/meta.yaml]: can you please remove the strict pinnings?

File: recipes/bam2fasta/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - bam2fasta
+
+  commands:
+    - bam2fasta --help
+    - bam2fasta info
+
+
+about:
+  home: https://github.com/czbiohub/bam2fasta
+  license: MIT License
```
Comment (bgruening) [inline recipes/bam2fasta/meta.yaml]: can you also add `license_file`

**Comment by pranathivemuri** [comment]
@bgruening Please review when you can! thanks

**Pull request merged**